modules/derivation/shared/src/main/scala/io/circe/derivation/DerivationMacros.scala

```diff
@@ -383,15 +865,35 @@ class DerivationMacros(val c: blackbox.Context) extends ScalaVersionCompat {
             }
         }
 
+
+
+        // we manage default serialization
         val fields: List[Tree] = repr.paramLists.flatten.map {
-          case Member(name, decodedName, tpe) =>
+          case Member(
+              name,
+              decodedName,
+              tpe,
+              keyName,
+              defaultValue,
+              noSerializeDefault
+              ) =>
             repr.encoder(tpe) match {
-              case Instance(_, _, instanceName) => q"""
-              _root_.scala.Tuple2.apply[_root_.java.lang.String, _root_.io.circe.Json](
-                ${transformName(decodedName)},
-                this.$instanceName.apply(a.$name)
-              )
-            """
+              case Instance(_, _, instanceName) =>
+                val realName = keyName.getOrElse(transformName(decodedName))
+
+                if (noSerializeDefault && defaultValue.isDefined) {
+                  q"""
+                    if(a.$name==${defaultValue.get}) None else
```

Comment:
Do you mind putting the prefixes on `None` and `Some` for the sake of hygiene?

Reply:
Could you better explain the change that you want? Witch prefix?

Reply:
The problem is that if the user has a different `None` definition at the macro use site, it'll get used instead of `scala.None`.